src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,161 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
                 if (!targetEntityGraphQLDetails.Enabled)
                 {
                     HandleOrRecordException(new DataApiBuilderException(
-                        message: $"entity: {relationship.TargetEntity} is disabled for GraphQL.",
+                        message: $"Entity: {relationship.TargetEntity} is disabled for GraphQL.",
                         statusCode: HttpStatusCode.ServiceUnavailable,
                         subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
                 }
 
+                // Linking object is null and therefore we have either a many to one or a one to many relationship. These relationships
+                // must be validated separately from many to many relationships. In one to many and many to one relationships, source
+                // fields and target fields need to match since they define the relationship between the two entities in its entirely.
+                // If both of these sets of fields are null, foreign key information will be used to define the relationship instead.
+                // see: https://learn.microsoft.com/en-us/azure/data-api-builder/relationships
+                if (relationship.LinkingObject is null)
```

Comment:
Should this be string.IsNullorWhitespace? e.g. `linkingObject: ""` wouldn't meet this condition, would it?